queries.sql

```diff
@@ -6,3 +6,85 @@ Select name, escape_attempts from animals where weight_kg > 10.5;
 Select * from animals where neutered = true;
 Select * from animals where name != 'Gabumon';
 select * from animals where weight_kg <= 17.3 and weight_kg >=10.4;
+
+-- Unspecified species
+Begin
+	Update animals 
+	Set species = 'unspecified';
+	Select * from animals;
+	Rollback;
+	
+Select * from animals;
+
+-- Species data
+Begin;
+	Update animals
+	Set species = 'digimon'
+	Where name like '%mon';
+
+	Update animals
+	Set species = 'pokemon'
+	Where name not like '%mon';
+
+	Select * from animals;
+	Commit;
+	
+Select * from animals;
+
+-- Rollback
+Begin;
+	Delete from animals;
+	Select * from animals;
+	Rollback;
+	
+Select * from animals;
+
+-- Savepoint
+Begin;
+	Delete from animals
+	Where date_of_birth > '2022-1-1';
+	Select * from animals;
+	Savepoint sp1;
+	
+	Update animals
+	Set weight_kg = weight_kg * -1;
+	Select * from animals;
+	Rollback to sp1;
+	
+	Select * from animals;
+	
+	Update animals
+	Set weight_kg = weight_kg * -1
+	Where weight_kg < 0;
+	
+	Select * from animals;
+Commit;
+
+Select * from animals;
+
+-- Aggregate functions and Group by
+Select count(id) from animals;
+Select count(*) as Home_owner from animals
+Where escape_attempts = 0;
+
+Select Avg(weight_kg) from animals;
+
+Select neutered, Count(*) as escape_count
```

Comment:
- [ ] Hello there! 👋 The current query is set up to return the count of escape attempts for animals, regardless of their neutered status. 📊 However, it would be even more insightful to retrieve the maximum number of escape attempts made by any animal, regardless of whether they are neutered or not. 📈 This modification will provide a clearer and more comprehensive view of the data, helping us identify the most escape-prone animal in our records. 🐾💥🔍

```suggestion
SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```